src/ScheduledJobs/Repositories/ParticipantRegistrationDynamoDbRepository.cs

```diff
@@ -24,14 +25,30 @@ public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamo
             _config = new DynamoDBOperationConfig { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
         }
 
-        public async Task<IEnumerable<Participant>> GetAllAsync()
+        public Task<IAsyncEnumerable<Participant>> GetAllAsync()
```

Comment:
If you are using `IAsyncEnumerable` might as well make it support cancellation and pass the token down through the stack:

`GetAllAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)`